src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -556,15 +572,30 @@ void RippleAddress::setAccountPrivate (
     setAccountPrivate (secretKey);
 }
 
-bool RippleAddress::accountPrivateSign (uint256 const& uHash, Blob& vucSig) const
+Blob RippleAddress::accountPrivateSign (Blob const& message) const
 {
-    vucSig = ECDSASign (uHash, getAccountPrivate());
-    const bool ok = !vucSig.empty();
+    if (vchData.size() == 33  &&  vchData[0] == 0xED)
+    {
+        uint8_t const*      secretKey = &vchData[1];
+        ed25519_public_key  publicKey;
+        Blob                signature (sizeof (ed25519_signature));
+
+        ed25519_publickey (secretKey, publicKey);
+
+        ed25519_sign (message.data(), message.size(), secretKey, publicKey, &signature[0]);
+
+        return signature;
+    }
+
+    uint256 const uHash = getSHA512Half (message);
+
+    Blob result = ECDSASign (uHash, getAccountPrivate());
+    const bool ok = !result.empty();
 
     CondLog (!ok, lsWARNING, RippleAddress)
```

Comment:
I think we should throw a fatal exception rather than logging. Rationale: logging at this low of a level doesn't make sense. It would be like `std::vector` logging. Plus, how can signing fail? We might want to audit that code to know what the conditions for failure are. I believe it leads back to OpenSSL bignum routines. which only fail when there is no memory.